Source/Languages/Casing/CaseStyleConverter.ts

```diff
@@ -0,0 +1,78 @@
+/// <reference path="CaseStyle.ts" />
+
+namespace GLS.Languages.Casing {
+    "use strict";
+
+    /**
+     * Allowed casing preferences.
+     */
+    export abstract class CaseStyleConverter {
+        /**
+         * Converts a name to the equivalent case style.
+         * 
+         * @param name   A name to convert.
+         * @returns The name's equivalent in this converter's case style.
+         */
+        public convert(name: string): string {
+            let position: number = 0;
+            let nameConverted: string = "";
+
+            while (position < name.length) {
+                let nextWordStart = this.findNextWordStart(name, position);
+                let nextWord = name.substring(position, nextWordStart);
+                nextWord = nextWord.replace(/_/g, "").toLowerCase();
```

Comment:
Removes all instances of `"_"`.

JavaScript's string::replace only removes the first instance when given a string, so you have to give it a regular expression to catch them all. Dumb language quirk.